Source/msg.cpp

```diff
@@ -979,8 +979,6 @@ DWORD OnSpellTile(const TCmd *pCmd, Player &player)
 		return sizeof(message);
 	if (message.wParam2 > RSPLTYPE_INVALID)
 		return sizeof(message);
-	if (message.wParam3 > static_cast<int>(Direction::SouthEast))
-		return sizeof(message);

```

Comment:
Are we not going to add any validation to the spell level? It seems it should have some basic range checks at least (`> 0` and  `< cap`)?

Reply:
Just saw this discussion:
- https://github.com/diasurgical/devilutionX/pull/3012#discussion_r720858392

I guess it resolves my comment.

Reply:
I think the max level is 20 + 2, but I don't feel confident enough to apply this restriction 😅

Reply:
What about `< 0`? Can spells with negative level be cast normally in the game somehow?

Reply:
`wParam3` is unsigned. So we shouldn't get negative numbers.